design/clusters.go

```diff
@@ -155,6 +155,38 @@ var _ = a.Resource("clusters", func() {
 		a.Response(d.InternalServerError, JSONAPIErrors)
 	})
 
+	a.Action("findByURL", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET("/"),
+		)
+		a.Params(func() {
+			a.Param("cluster_url", d.String, "the URL of the cluster to show")
+			a.Required("cluster_url")
+		})
+		a.Description("Get single cluster configuration given its URL")
+		a.Response(d.OK, showSingleCluster)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+	})
+
+	a.Action("findByURLForAuth", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET("/auth"),
+		)
+		a.Params(func() {
+			a.Param("cluster_url", d.String, "the URL of the cluster to show")
+			a.Required("cluster_url")
+		})
+		a.Description("Get single cluster configuration given its URL, with full info")
+		a.Response(d.OK, showSingleCluster)
+		a.Response(d.Unauthorized, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
```

Comment:
I think we are missing `400 Bad Request`? e.g. cluster URL foo.com which is not a valid cluster url.